bundles/CoreBundle/Controller/UserAwareController.php

```diff
@@ -0,0 +1,272 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\CoreBundle\Controller;
+
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Pimcore\Security\User\User as UserProxy;
+use Pimcore\Controller\Controller;
+use Pimcore\Extension\Bundle\PimcoreBundleManager;
+use Pimcore\Logger;
+use Pimcore\Model\User;
+use Pimcore\Security\User\TokenStorageUserResolver;
+use Symfony\Component\HttpKernel\Event\ControllerEvent;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\Serializer\Encoder\DecoderInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Translation\Exception\InvalidArgumentException;
+use Symfony\Contracts\Service\Attribute\Required;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+abstract class UserAwareController extends Controller
```

Comment:
Should we maybe put that outside `CoreBundle` along with the `FrontendController` which is in `Pimcore\Controller\...`?

Reply:
yes, we could do this since it is not based on any CoreBundle class.